### ISSUE TITLE
Variable substitution missed out

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#445]
```
?- {"c",b,a} = {[X|_],Y,Z}, D = {[X,Y],Z}.
   X = c, Y = b, Z = a, D = {"cb",Z}. % Z instead of a
```

**Comment by flexoron** [comment]
see #444

**Comment by flexoron** [comment]
v2.32.5
```
?- {"c",b,a} = {[X|_],Y,Z}, D = {[X,Y],Z}.
   X = c, Y = b, Z = a, D = {"cb",a}.

?- {"c",b,a} = {[X|_],Y,Z}, D = {[X,Y],Z,Z}.
   X = c, Y = b, Z = a, D = {"cb",Z,a}.
```
```
Z is the wrongdoer

?- {"c",b,a} = {[X|_],Y,Z}, D = {[X,Y],X,X,X}.
   X = c, Y = b, Z = a, D = {"cb",c,c,c}.

?- {"c",b,a} = {[X|_],Y,Z}, D = {[X,Y],Y,Y,Y}.
   X = c, Y = b, Z = a, D = {"cb",b,b,b}.

?- {"c",b,a} = {[X|_],Y,Z}, D = {[X,Y],Z,Z,Z}.
   X = c, Y = b, Z = a, D = {"cb",Z,Z,a}.
```

**Comment by flexoron** [comment]
Fixed with v2.32.6
